src/monodroid/jni/monodroid-glue.cc

```diff
@@ -1582,23 +1581,10 @@ MonodroidRuntime::set_profile_options ()
 		value.assign (prop_value);
 	}
 
+#if defined (NET6)
 	// setenv(3) makes copies of its arguments
 	setenv ("DOTNET_DiagnosticPorts", value.get (), 1);
-}
-#else // def NET6
-inline void
-MonodroidRuntime::set_profile_options ()
-{
-	// We want to avoid dynamic allocation, thus let’s create a buffer that can take both the property value and a
-	// path without allocation
-	dynamic_local_string<SENSIBLE_PATH_MAX + PROPERTY_VALUE_BUFFER_LEN> value;
-	{
-		dynamic_local_string<PROPERTY_VALUE_BUFFER_LEN> prop_value;
-		if (androidSystem.monodroid_get_system_property (Debug::DEBUG_MONO_PROFILE_PROPERTY, prop_value) == 0)
-			return;
-
-		value.assign (prop_value.get (), prop_value.length ());
-	}
+#endif // def NET6
```

Comment:
Could we instead copy the code that deals with the `aotprofile` option to the net6 version of `set_profile_options`? None of the other options work in net6+. Copying the code will make it clear what is and what isn't supported in net6 and will also make it easier to just delete the "legacy" option once we drop the support.

Reply:
I think I did it right? Let me know if there is code that can be removed.